bin/templates/cordova/lib/retry.js

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env node
+
+/*
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+/* jshint node: true */
+
+"use strict";
+
+/*
+ * Retry a promise-returning function attemts_left times, propagating its
+ * results on success or throwing its error on a failed final attempt.
+ *
+ * Takes a number of attempts, a promise-returning function, and all its
+ * arguments (as trailing arguments).
+ */
+module.exports.retryPromise = function (attemts_left, promiseFunction) {
```

Comment:
Consider writing unit tests for this one as it is fairly easy and isolated to write for this function.